frontend/src/pages/modelServing/screens/global/ModelServingGlobal.stories.tsx

```diff
@@ -143,3 +143,58 @@ DeleteModel.play = async ({ canvasElement }) => {
   await userEvent.type(retypeNameInput, 'Test Inference Service', { delay: 50 });
   expect(deleteButton).not.toBeDisabled();
 };
+
+export const DeployModel = Template.bind({});
+DeployModel.parameters = {
+  a11y: {
+    // need to select modal as root
+    element: '.pf-c-backdrop',
+  },
+};
+DeployModel.play = async ({ canvasElement }) => {
+  // load page and wait until settled
+  const canvas = within(canvasElement);
+  await canvas.findByText('Test Inference Service', undefined, { timeout: 5000 });
+
+  // user flow for editing a project
+  await userEvent.click(canvas.getByText('Deploy model', { selector: 'button' }));
+
+  // get modal
+  const body = within(canvasElement.ownerDocument.body);
+  const nameInput = body.getByRole('textbox', { name: 'Model Name' });
+  const updateButton = body.getByText('Deploy', { selector: 'button' });
+
+  // test that you can not submit on empty
+  await userEvent.clear(nameInput);
+  expect(updateButton).toBeDisabled();
+
+  // fill the form
+  await userEvent.type(nameInput, 'Model Name', { delay: 50 });
+  body.getAllByRole('button', { name: 'Options menu' });
+  await userEvent.click(body.getAllByRole('button', { name: 'Options menu' })[0]);
+  await userEvent.click(body.getByText('Test Project', { selector: 'button' }));
+  await body.findByText('Select a framework');
+  await userEvent.click(body.getAllByRole('button', { name: 'Options menu' })[1]);
+  await userEvent.click(body.getByText('onnx - 1'));
+  await body.findByText('Select...');
+  await userEvent.click(body.getAllByRole('button', { name: 'Options menu' })[2]);
+  await body.findByText('Test Secret');
+  await userEvent.click(body.getByText('Test Secret'));
+  expect(updateButton).not.toBeDisabled();
+
+  await userEvent.click(body.getByRole('radio', { name: 'New data connection' }));
```

Comment:
Any reason we do both in one modal test? I guess since we don't submit it's not a problem -- but it will prevent us from stacking tests later when we want to try to do e2e stuff.